app/views/home/opensource.html.erb

```diff
@@ -1,5 +1,91 @@
   <div class="container cointaier-shrinked">
-    <h1 class="display-4 text-left">OpenSource</h1>
+    <h1 class="display-4 text-left text-success">OpenSource Projects</h1>
+    <h2 class="text-muted text-left">We offer free hosting to selected open source projects.</h2>
   </div>
-</section>
\ No newline at end of file
+</section>
+<section class="pt-8 pt-md-11 bg-light p-0 p-md-5">
+  <div class="container carousel-opensource d-none d-md-block"> 
+    <div id="carouselIndicators" class="carousel slide d-flex h-100" data-ride="carousel">
+      <ol class="carousel-indicators">
+        <% @featured_projects.each_with_index do |project,index| %>   
+          <li data-target="#carouselIndicators"
+              data-slide-to="<%= index %>"
+              class="<%= index == 0 ? 'active' : '' %>">
+          </li>
+        <% end %>
+      </ol>
+      <div class="carousel-inner">
+        <% @featured_projects.each_with_index do |project,index| %>
+          <div class="carousel-item p-5 <%= index == 0 ? 'active' : '' %>"
+               style="background:<%= project[:image][:background] %>">
+            <div class="d-flex h-100">
+              <div class="text-center item-img align-self-center m-auto">
+                <img class="d-block m-auto align-middle"
+                    width="<%= project[:image][:width] %>"
+                    src="<%= project[:image][:src] %>" 
+                    alt="<%= project[:name] %>"/>
+              </div>
+              <div class="carousel-caption d-none d-md-block">
+                <p class="text-<%= project[:textClass] %>"><%= project[:description] %></p>
+              </div>                
+            </div>          
+          </div>
+        <% end %>
+      </div>
+      <a class="carousel-control-prev"
+         href="#carouselIndicators"
+         role="button"
+         data-slide="prev">
+        <i class="fa fa-2x fa-chevron-left text-secondary"
+              aria-hidden="true">
+        </i>
+        <span class="sr-only">Previous</span>
+      </a>
+      <a class="carousel-control-next"
+         href="#carouselIndicators"
+         role="button"
+         data-slide="next">
+        <i class="fa fa-2x fa-chevron-right text-secondary" aria-hidden="true"></i>
+        <span class="sr-only">Next</span>
+      </a>
+    </div>
+  </div>
+</section>
+<section class="opensource-list pt-8 pt-md-11 bg-secondary p-3 p-md-5">
+  <div class="container">
+    <div class="row">
+    <% @opensource_projects.each do |project| %>
+      <div class="col-md-4">
+        <div class="card mb-4 shadow-sm hvr-grow">
+          <div class="card-header"
+               style="background:<%= project[:image][:background] %>">
+              <a href="/opensource/<%= project[:slug] %>" class="d-flex h-100">
+                <img class="d-block w-50 m-auto align-self-center"
+                    src="<%= project[:image][:src] %>" 
```

Comment:
the image should be optional, don't think everyone will have a logo. And so when there is no image the name should probably be bigger (?)

Reply:
Done. Added a conditional for the logo. Displaying the name if the logo src is blank.